internal/tuf/v02/root.go

```diff
@@ -467,6 +467,24 @@ func (r *RootMetadata) AddPropagationDirective(directive tuf.PropagationDirectiv
 	return nil
 }
 
+// Updatepropagation checks if there is a duplicate or update and deal with that
+func (r *RootMetadata) UpdatePropagationDirective(directive tuf.PropagationDirective) error {
+	//find if there is a duplicate(exact same) or update(same name, different in other)
+	// in the slice in linear search, then update the found propagation
+	for i := range r.Propagations {
+		if r.Propagations[1].GetName() == directive.GetName() {
+			r.Propagations[i] = directive
+			return nil
+		}
+	}
+
+	// if the directive is new, call AddPropagationDirective to deal with it
+	if err := r.AddPropagationDirective(directive); err != nil {
+		return err
+	}
+	return nil
```

Comment:
Same as in `internal/tuf/v01/root.go`.

```suggestion
return ErrPropagationDirectiveNotFound
```